packages/skills/weather_client/handlers.py

```diff
@@ -86,100 +98,142 @@ def teardown(self) -> None:
-        """Initialise the oef handler."""
-        super().__init__(**kwargs)
-        self.dialogue_id = 1
+    def _handle_decline(self, msg: FIPAMessage, sender: str, message_id: int, dialogue_id: int, dialogue: Dialogue) -> None:
+        """
+        Handle the decline.
 
-    def setup(self) -> None:
-        """Call to setup the handler."""
-        pass
+        :param msg: the message
+        :param sender: the sender
+        :param message_id: the message id
+        :param dialogue_id: the dialogue id
+        :param dialogue: the dialogue object
+        :return: None
+        """
+        logger.info("[{}]: received DECLINE from sender={}".format(self.context.agent_name, sender[-5:]))
 
-    def handle(self, message: Message, sender: str) -> None:
+    def _handle_match_accept(self, msg: FIPAMessage, sender: str, message_id: int, dialogue_id: int,
+                             dialogue: Dialogue) -> None:
         """
-        Implement the reaction to a message.
+        Handle the match accept.
 
-        :param message: the message
+        :param msg: the message
         :param sender: the sender
+        :param message_id: the message id
+        :param dialogue_id: the dialogue id
+        :param dialogue: the dialogue object
         :return: None
         """
-        msg_type = OEFMessage.Type(message.get("type"))
-
-        if msg_type is OEFMessage.Type.SEARCH_RESULT:
-            agents = cast(List[str], message.get("agents"))
-            logger.info("[{}]: found agents={}".format(self.context.agent_name, agents))
-            for agent in agents:
-                msg = FIPAMessage(message_id=STARTING_MESSAGE_ID,
-                                  dialogue_id=self.dialogue_id,
-                                  performative=FIPAMessage.Performative.CFP,
-                                  target=STARTING_TARGET_ID,
-                                  query=None
-                                  )
-                self.dialogue_id += 1
-                self.context.outbox.put_message(to=agent,
-                                                sender=self.context.agent_public_key,
-                                                protocol_id=FIPAMessage.protocol_id,
-                                                message=FIPASerializer().encode(msg))
+        fipa_msg = cast(FIPAMessage, dialogue.last_incoming_message)
+        new_message_id = cast(int, fipa_msg.get("message_id")) + 1
+        new_target_id = cast(int, fipa_msg.get("target")) + 1
+        dialogue_id = cast(int, fipa_msg.get("dialogue_id"))
+        counterparty_pbk = dialogue.dialogue_label.dialogue_opponent_pbk
+        inform_msg = FIPAMessage(message_id=new_message_id,
+                                 dialogue_id=dialogue_id,
+                                 target=new_target_id,
+                                 performative=FIPAMessage.Performative.INFORM,
+                                 json_data={"transaction": "Done"})
```

Comment:
Could add as message rather than `Done`: `Sending payment via bank transfer`.

The idea is that the payment is done off chain.